src/domain/individualApplication/IndividualApplicationPageContainer.tsx

```diff
@@ -51,17 +57,58 @@ const IndividualCustomerPageContainer: React.SFC = () => {
     },
   });
 
-  const { data: customersData } = useQuery<
-    FILTERED_CUSTOMERS,
-    FILTERED_CUSTOMERS_VARS
-  >(FILTERED_CUSTOMERS_QUERY, {
-    variables: {
-      lastName: data?.berthApplication?.lastName,
-    },
+  const [debouncedSearchVal] = useDebounce(searchVal, 500, {
+    equalityFn: (prev, next) => prev === next,
   });
 
+  const [
+    fetchFilteredCustomers,
+    { data: customersData, refetch, called },
+  ] = useLazyQuery<FILTERED_CUSTOMERS, FILTERED_CUSTOMERS_VARS>(
+    FILTERED_CUSTOMERS_QUERY,
+    {
+      variables: {
+        [searchBy]: debouncedSearchVal ?? searchVal,
+      },
+    }
+  );
+
   // TODO: handle errors
   const [deleteDraftedApplication] = useDeleteBerthApplication();
+  useEffect(() => {
+    if (!data?.berthApplication) return;
+    let initialSearchBy: keyof BerthApplication;
+
+    switch (searchBy) {
+      case SearchBy.EMAIL:
+        initialSearchBy = 'email';
+        break;
+      case SearchBy.ADDRESS:
+        initialSearchBy = 'address';
+        break;
+      case SearchBy.FIRST_NAME:
+        initialSearchBy = 'firstName';
+        break;
+      default:
+        initialSearchBy = 'lastName';
+        break;
+    }
+
+    setSearchVal(data.berthApplication[initialSearchBy]);
```

Comment:
Not really a comment on the code, just a question about the business logic here.

Whose data is being set here as the initial value if the application is not connected to anybody?

Reply:
Good question! 
If the application doesn't have a customer, it means that it's not yet linked to an existing customer in our database, but still, it has the personal info of the one who submitted it, and we populate the initial values of the search field with this personal info.

Reply:
That makes sense. Thanks for the clarification :+1: